components/esp_netif/lwip/esp_netif_lwip_slip.c

```diff
@@ -0,0 +1,335 @@
+    //! ESP netif object
+    esp_netif_t *esp_netif;
+
+    ip6_addr_t addr;      /* Local IP6 address (prefix derived from this) */
+
+    //! Underlying uart port configuration
+    uart_port_t uart_dev;
+    uint32_t uart_baud;
+    int uart_tx_pin;
+    int uart_rx_pin;
+
+    QueueHandle_t uart_queue;
+    TaskHandle_t rx_task;
+
+    uint8_t* buffer;
+    uint32_t buffer_len;
+
+    slip_rx_filter_f* rx_filter;
+    void* rx_filter_ctx;
+
+    bool running;
+} lwip_slip_ctx_t;
+
+
+static void esp_slip_uart_rx_task(void *arg);
+
+
+/**
+ * @brief Create a new lwip slip interface
+ */
+lwip_slip_ctx_t* esp_netif_new_slip(esp_netif_t *esp_netif, const esp_netif_netstack_config_t *esp_netif_stack_config) {
+    ESP_LOGD(TAG, "%s", __func__);
+
+    // Fetch netif and create context
+    struct netif * netif_impl = esp_netif->lwip_netif;
+
+    lwip_slip_ctx_t * slip_ctx = calloc(1, sizeof(lwip_slip_ctx_t));
+    if (slip_ctx == NULL) {
+        ESP_LOGE(TAG, "%s: cannot allocate lwip_slip_ctx_t", __func__);
+        return NULL;
+    }
+
+    ESP_LOGD(TAG, "%s: Initialising SLIP (esp_netif %p, lwip_netif %p, device: %d)", __func__, esp_netif, netif_impl, slip_ctx->uart_dev);
+
+    // Load config
+    const esp_netif_slip_config_t* slip_config = &esp_netif_stack_config->lwip_slip.slip_config;
+
+    slip_ctx->uart_dev = slip_config->uart_dev;
+    slip_ctx->buffer_len = slip_config->rx_buffer_len;
+    slip_ctx->buffer = NULL;
+
+    // Attache network interface
+    slip_ctx->esp_netif = esp_netif;
+
+#if 0
+    // Initialise slip API
+    int32_t res = slipif_init(netif_impl);
+    if (res != ERR_OK) {
+        ESP_LOGE(TAG, "%s: error %d initialising slipif", __func__, res);
+        
+        free(slip_ctx);
+
+        return NULL;
+    }
+#endif
+
+    ESP_LOGI(TAG, "%s: Created SLIP interface (netif %p, slip_ctx: %p)", __func__, esp_netif, slip_ctx);
+
+    return slip_ctx;
+}
+
+/**
+ * @brief Creates new SLIP related structure
+ */
+esp_err_t esp_netif_start_slip(lwip_slip_ctx_t *slip_ctx) {
+    ESP_LOGI(TAG, "%s: Starting SLIP connection (slip_ctx: %p, addr: %s)", __func__, slip_ctx, ip6addr_ntoa(&slip_ctx->addr));
+
+    struct netif * netif_impl = slip_ctx->esp_netif->lwip_netif;
+
+    // Store serial port number in net interface for SIO open command
+    netif_impl->state = (void*)slip_ctx->uart_dev;
+
+    // Allocate RX buffer if not existing
+    if (slip_ctx->buffer == NULL) {
+        slip_ctx->buffer = malloc(slip_ctx->buffer_len);
+    }
+
+#if 1
+    // Attach interface
+    netif_add_noaddr(slip_ctx->esp_netif->lwip_netif, (void*)slip_ctx->uart_dev, &slipif_init, &ip_input);
+
+    // Bind address
+    int8_t addr_index = 0;
+    //netif_add_ip6_address(slip_ctx->esp_netif->lwip_netif, &slip_ctx->addr, &addr_index);
+
+    netif_ip6_addr_set(slip_ctx->esp_netif->lwip_netif, addr_index, &slip_ctx->addr);
+    netif_ip6_addr_set_state(slip_ctx->esp_netif->lwip_netif, addr_index, IP6_ADDR_VALID);
```

Comment:
```c
int8_t addr_index;
netif_add_ip6_address(slip_ctx->esp_netif->lwip_netif, &slip_ctx->addr, &addr_index);
netif_ip6_addr_set_state(slip_ctx->esp_netif->lwip_netif, addr_index, IP6_ADDR_VALID);
```

[`netif_ip6_addr_set`](http://www.nongnu.org/lwip/2_0_x/group__netif__ip6.html#gae21572fdbd8664d22a1b281a6c31c9bbf) is used to change an existing address, `netif_add_ip6_address` could do the trick

Reply:
When I tested it calling `netif_ip6_addr_set` resulted in the address state being set to `IP6_ADDR_TENTATIVE` and then never upgraded so it didn't work... not sure why this is

Reply:
I think using the `netif_ip6_addr_set()` is okay in this context as we actually set an IP before connecting bypassing all the IPv6 negotiation/ND.